### PR TITLE
Docs: upgrade MetalLB documentation website

### LINKED ISSUE #2362: Fix Hugo CI mismatch version & Upgrade Hugo to the latest version
### Is your feature request related to a problem?

I noticed that the Hugo version of GitHub Actions does not match Netlify.

GitHub Actions: https://github.com/metallb/metallb/blob/fe385845877a9ef48ef6b519896854e7ca5c4615/.github/workflows/site-build.yaml#L28
Netlify: https://github.com/metallb/metallb/blob/fe385845877a9ef48ef6b519896854e7ca5c4615/netlify.toml#L7

However, Hugo 0.32 is now 7 years old, we should upgrade this latest version.

### Describe the solution you'd like

1. Upgrade Hugo to the latest version
2. Fix build errors and warning messages

### Additional context

This issue has been discussed with the maintainer and agreed to be implemented.

https://kubernetes.slack.com/archives/CTU63S7K3/p1712581823970119?thread_ts=1712054855.096469&cid=CTU63S7K3

### I've read and agree with the following

- [X] I've checked all open and closed issues and my request is not there.
- [X] I've checked all open and closed pull requests and my request is not there.

### PR DESCRIPTION
**Pull request by tico88612**
**Is this a BUG FIX or a FEATURE ?**:

/kind documentation

**What this PR does / why we need it**:
- Resolve #2362 

**Special notes for your reviewer**:
If it's not convenient to preview the content locally, I can set up a demo site first.

**Release note**:

```release-note
- Upgrade Hugo to v0.124.1
- Theme change to `Hugo Relearn Theme` (version: 5.27.0)
- Update config.toml
```

### CODE REVIEW

File: website/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# MetalLB Documentation Website
```
Comment (fedepaol) [inline website/README.md:1]: I'd put this under the "contributing" section of the website, with the suggestion that any new feature should be documented

Comment (fedepaol) [reply website/README.md:1]: I think the discoverability of an inner README is quite low :-)

Now that I think about it, I'd rather add yet another inv command (something like `inv preview-website`) that downloads hugo locally and previews the website, so a new contributor won't even have the burden of finding the right version, checking how to build it and to run it.

And we don't need to do it here, I'll file a separate issue but I think it's the right path. On the doc page we can just say "run inv preview" and check if your changes are correct.

Comment (tico88612) [reply website/README.md:1]: I think it's better that way.

**Review by cyclinder — APPROVED** [review]
I also tried it locally, looks good! thanks!

LGTM

File: website/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Please follow Hugo's official website documentation to install.
+
+The currently used version of Hugo is defined in the [`netlify.toml`](../netlify.toml) configuration file.
+
+## Running the site locally
+
+If you want to develop the site locally, you can run a single command (assuming that you've run the [setup](#setup)):
+
+```bash
+$ make develop
```
Comment (fedepaol) [inline website/README.md:14]: we don't use make here (unless I am missing something). As I wrote above, I'd remove this README and extend the website contributing guidelines when we implement https://github.com/metallb/metallb/issues/2382

Comment (tico88612) [reply website/README.md]: Removed this README.md.

**Pull request merged**